test/shopify_app/scripttags_manager_test.rb

```diff
@@ -0,0 +1,83 @@
+require 'test_helper'
+
+class ShopifyApp::ScripttagsManagerTest < ActiveSupport::TestCase
+
+  setup do
+    ShopifyApp.configure do |config|
+      config.scripttags = [
+        {event: 'onload', src: 'https://example-app.com/fancy.js'},
+        {event: 'onload', src: 'https://example-app.com/foobar.js'}
+      ]
+    end
+
+    @manager = ShopifyApp::ScripttagsManager.new("regular-shop.myshopify.com", "token")
+  end
+
+  test "#create_scripttags makes calls to create scripttags" do
+    ShopifyAPI::ScriptTag.stubs(all: [])
+
+    expect_scripttag_creation('onload', 'https://example-app.com/fancy.js')
+    expect_scripttag_creation('onload', 'https://example-app.com/foobar.js')
+    @manager.create_scripttags
+  end
+
+  test "#create_scripttags when creating a scripttag fails, raises an error" do
+    ShopifyAPI::ScriptTag.stubs(all: [])
+    scripttag = stub(persisted?: false, errors: stub(messages: ["Source needs to be https"]))
+    ShopifyAPI::ScriptTag.stubs(create: scripttag)
+
+    assert_raise ShopifyApp::ScripttagsManager::CreationFailed do
+      @manager.create_scripttags
+    end
+  end
+
+  test "#create_scripttags when creating a scripttag fails and the scripttag exists, do not raise an error" do
+    scripttag = stub(persisted?: false)
```

Comment:
Sorry for replying long ago - I guess I didn't see the messages.
I've removed this test because it doesn't actually make any sense - you would never actually create a scripttag if it already exists - there's a check to make sure of that. And if it did fail, it should raise an error.